core/src/main/resources/org/apache/spark/ui/static/webui.css

```diff
@@ -194,6 +194,24 @@ span.additional-metric-title {
   color: #777;
 }
 
+#additional-metrics {
+  float: left;
+  padding-bottom: 10px;
+}
+
+#links {
+  float: left;
+  padding-left: 100px;
```

Comment:
My understanding has always been that using `<table>`s to lay out things that are not tabular but that you merely want to be horizontally adjacent is hacky and should be avoided. If you're adamant I can do it though.

I triedy `display: inline` since that is maybe supposed to directly address this kind of use case:

![](https://s3.amazonaws.com/f.cl.ly/items/1U19471V133N0l1l2y3R/Screen%20Shot%202015-03-21%20at%206.40.22%20PM.png)

that lets me kill the `clear:both` on the "Summary metrics …" &lt;h4&gt;, but it gets in to trouble when I expand the "additional metrics":

![](https://s3.amazonaws.com/f.cl.ly/items/05272a1r3P240N0N1D1r/Screen%20Shot%202015-03-21%20at%206.41.18%20PM.png)

re: adding it under the title, it would be nice to not have this push the actual important info on the page further below the fold. There is a preponderance of unused horizontal space on the page currently, and the row it is being added to is a fairly underutilized one since the "additional metrics" dropdown is presumably not frequently engaged with.

Here it is full-screen on my macbook:

![](https://s3.amazonaws.com/f.cl.ly/items/1C2z0m2y3J2E2O0l020o/Screen%20Shot%202015-03-21%20at%206.48.07%20PM.png)

The "Summary Metrics …" &lt;h4&gt; is 31% of the way down the screen.

With "additional metrics" open it is 50% of the way down:

![](https://s3.amazonaws.com/f.cl.ly/items/1Q1M3c0F0o3g0u1U0H1S/Screen%20Shot%202015-03-21%20at%206.49.29%20PM.png)